File: frontend/src/types/referenceData.ts
export type ReferenceDataInputType = "text" | "textarea" | "url" | "badge_preset";

export interface ReferenceDataField {
  key: string;
  label: string;
  input_type: ReferenceDataInputType;
  required: boolean;
  show_in_list: boolean;
  placeholder: string | null;
  help_text: string | null;
}

export interface ReferenceDataResource {
  key: string;
  label: string;
  plural_label: string;
  description: string;
  api_path: string;
  settings_path: string;
  search_fields: string[];
  fields: ReferenceDataField[];
}

export interface ReferenceDataRecord {
  id: string;
  [key: string]: string | null | undefined;
}

File: backend/app/schemas/reference_data.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


ReferenceFieldType = Literal["text", "textarea", "url", "badge_preset"]


class ReferenceDataFieldRead(BaseModel):
    key: str
    label: str
    input_type: ReferenceFieldType
    required: bool = False
    show_in_list: bool = True
    placeholder: str | None = None
    help_text: str | None = None


class ReferenceDataResourceRead(BaseModel):
    key: str
    label: str
    plural_label: str
    description: str
    api_path: str
    settings_path: str
    search_fields: list[str]
    fields: list[ReferenceDataFieldRead]
